scipy/stats/_multicomp.py

```diff
@@ -24,10 +27,166 @@
 class DunnettResult:
     statistic: np.ndarray
     pvalue: np.ndarray
+    _alternative: Literal['two-sided', 'less', 'greater'] = field(repr=False)
+    _rho: np.ndarray = field(repr=False)
+    _df: int = field(repr=False)
+    _std: float = field(repr=False)
+    _mean_samples: np.ndarray = field(repr=False)
+    _mean_control: np.ndarray = field(repr=False)
+    _n_samples: np.ndarray = field(repr=False)
+    _n_control: int = field(repr=False)
+    _rng: SeedType = field(repr=False)
+    _ci: ConfidenceInterval | None = field(default=None, repr=False)
+    _ci_cl: DecimalNumber | None = field(default=None, repr=False)
+
+    def __str__(self):
+        # Note: `__str__` prints the confidence intervals from the most
+        # recent call to `confidence_interval`. If it has not been called,
+        # it will be called with the default CL of .95.
+        if self._ci is None:
+            self.confidence_interval(confidence_level=.95)
+        s = (
+            "Dunnett's test"
+            f" ({self._ci_cl*100:.1f}% Confidence Interval)\n"
+            "Comparison               Statistic  p-value  Lower CI  Upper CI\n"
+        )
+        for i in range(self.pvalue.size):
+            s += (f" (Sample {i} - Control) {self.statistic[i]:>10.3f}"
+                  f"{self.pvalue[i]:>10.3f}"
+                  f"{self._ci.low[i]:>10.3f}"
+                  f"{self._ci.high[i]:>10.3f}\n")
+
+        if self._alternative == 'greater':
+            s += (
+                "\nOne-sided alternative (greater): sample i's mean exceed "
+                "the control's mean by an amount at least Lower CI"
+            )
+        elif self._alternative == 'less':
+            s += (
+                "\nOne-sided alternative (less): sample i's mean exceed "
+                "the control's mean by an amount at most Upper CI"
+            )
+        else:
+            s += (
+                "\nTwo-sided alternative: sample i's mean exceed the "
+                "control's mean by an amount between Lower CI and Upper CI"
+            )
```

Comment:
Rather than wordsmith these, I'd like to remove them on the following basis:
- If someone already understands confidence intervals, they will be able to interpret the numerical results appropriately without these. For example, if we read that `Sample 0 - Conrol` is as `low` as `-np.inf` and `high` as 10, we can translate this into an English statement like "sample 0's mean exceeds the control's mean by an amount at most 10".
- If someone does not understand confidence intervals, they will not be able to interpret the numerical results appropriately with or without these. These are stated as if they are facts rather than an inference with some amount of confidence (let alone describe what "confidence" really means, which is more complicated).